kodilla-testing/src/main/java/com/kodilla/testing/library/BookLibrary.java

```diff
@@ -0,0 +1,31 @@
+package com.kodilla.testing.library;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BookLibrary {
+
+    private LibraryDatabase libraryDatabase;
+
+    public BookLibrary(LibraryDatabase libraryDatabase) {
+        this.libraryDatabase = libraryDatabase;
+    }
+
+    public List<Book> listBooksWithCondition(String titleFragment) {
+
+        List<Book> bookList = new ArrayList<Book>();
+        if (titleFragment.length() < 3) return bookList;
+
+        List<Book> resultList = libraryDatabase
+                .listBooksWithCondition(titleFragment);
+        if (resultList.size() > 20) return bookList;
+
+        bookList = resultList;
```

Comment:
w sumie to nie potrzebujesz zmiennej bookList. Mógłbyś zawsze zwracać w tych przypadkach` Collections.emptyList()` albo poprostu `new ArrayList<>()`